library/src/main/java/eightbitlab/com/blurview/DefaultBlurController.java

```diff
@@ -204,64 +203,72 @@ private void drawUnderlyingViews() {
         rootView.draw(internalCanvas);
     }
 
-    @Override
-    public void drawBlurredContent(Canvas canvas) {
+    @Override public void drawBlurredContent(Canvas canvas) {
         isMeDrawingNow = true;
 
         internalCanvas.save();
         setupInternalCanvasMatrix();
         drawUnderlyingViews();
         internalCanvas.restore();
 
-        blurAndSave();
-        draw(canvas);
+        if (enabled) {
+            blurAndSave();
+            draw(canvas);
+        }
     }
 
     private void draw(Canvas canvas) {
         canvas.save();
-        canvas.scale(scaleFactor * roundingWidthScaleFactor, scaleFactor * roundingHeightScaleFactor);
+        canvas.scale(scaleFactor * roundingWidthScaleFactor,
+                scaleFactor * roundingHeightScaleFactor);
         canvas.drawBitmap(internalBitmap, 0, 0, null);
         canvas.restore();
     }
 
-    @Override
-    public void onDrawEnd(Canvas canvas) {
+    @Override public void onDrawEnd(Canvas canvas) {
         blurView.post(onDrawEndTask);
     }
 
     private void blurAndSave() {
         internalBitmap = blurAlgorithm.blur(internalBitmap, blurRadius);
     }
 
-    @Override
-    public void updateBlurViewSize() {
+    @Override public void updateBlurViewSize() {
         int measuredWidth = blurView.getMeasuredWidth();
         int measuredHeight = blurView.getMeasuredHeight();
 
         init(measuredWidth, measuredHeight);
     }
 
-    @Override
-    public void destroy() {
+    @Override public void destroy() {
         stopAutoBlurUpdate();
         blurAlgorithm.destroy();
         if (internalBitmap != null) {
             internalBitmap.recycle();
         }
     }
 
-    @Override
-    public void setBlurRadius(float radius) {
+    @Override public void enabledOnStart(boolean enabled) {
```

Comment:
I thought if the user wants to remove blur effect on start and maybe add later. Is that ok?